Source/Core/InputCommon/ControllerInterface/SDL/SDL.cpp

```diff
@@ -187,30 +187,55 @@ class GameController : public Core::Device
   };
 
   // Rumble
-  template <int LowEnable, int HighEnable, int SuffixIndex>
-  class GenericMotor : public Output
+  class Motor : public Output
   {
   public:
-    explicit GenericMotor(SDL_GameController* gc) : m_gc(gc) {}
-    std::string GetName() const override
+    Motor(const char* name, GameController& gc, Uint16* state)
+        : m_name{name}, m_gc{gc}, m_state{*state}
     {
-      return std::string("Motor") + motor_suffixes[SuffixIndex];
     }
+    std::string GetName() const override { return m_name; }
     void SetState(ControlState state) override
     {
-      Uint16 rumble = state * std::numeric_limits<Uint16>::max();
-      SDL_GameControllerRumble(m_gc, rumble * LowEnable, rumble * HighEnable, RUMBLE_LENGTH_MS);
+      const auto new_state = state * std::numeric_limits<Uint16>::max();
+      if (m_state == new_state)
+        return;
+
+      m_state = new_state;
+      m_gc.UpdateRumble();
     }
 
   private:
-    SDL_GameController* const m_gc;
+    const char* const m_name;
```

Comment:
Can the name be a `string_view` since it is expected to be a literal?

Reply:
I suppose, but I don't know if that would gain us anything. And I wouldn't want to make the constructor take a `string_view` since that might not make it quite as obvious at the call site that the `string_view` outlives the call. One could inadvertently pass a `std::string` r-value and it would create a dangling `string_view`.